v1/util/cloud_events_util.cc

```diff
@@ -0,0 +1,66 @@
+#include "cloud_events_util.h"
+
+#include <google/protobuf/util/time_util.h>
+
+namespace cloudevents {
+namespace util {
+
+using ::io::cloudevents::v1::CloudEvent;
+using ::io::cloudevents::v1::CloudEvent_CloudEventAttribute;
+
+bool CloudEventsUtil::IsValid(CloudEvent cloud_event) {
+    return !(cloud_event.id().empty() || 
+        cloud_event.source().empty() || 
+        cloud_event.spec_version().empty() ||
+        cloud_event.type().empty());
+}
+
+absl::StatusOr<std::map<std::string, CloudEvent_CloudEventAttribute>> CloudEventsUtil::GetMetadata(
```

Comment:
Is this supposed to be unordered_map? Either way, use absl::flat_hash_map if possible. Advantages see https://abseil.io/docs/cpp/guides/container#hash-tables.